### DISCUSSION TITLE
Setting up workflows with many possible parameter values

### DISCUSSION THREAD

**Discussion by Kodiologist** [ropensci/targets#946] [Help] [answered]
I have a `targets` workflow that uses static branches to specify a region of the earth, and I'm thinking of extending it to cover larger sets of regions, say, a country of the user's choice; the code would select geometry for that country from some source of per-country shapefiles. But this could lead to a very large number of branches existing even though the user might only want to make one of them, or a handful of them. So then dynamic branches look like a better choice than static branches, but from [the chapter on dynamic branching in the manual](https://books.ropensci.org/targets/dynamic.html#branching-over-files), I don't see how to `tar_make` only a branch of one's choice. `tar_read` has a `branches` parameter, but not `tar_make`.

More generally, you could imagine a target with one or more parameters that could take a very large number of values, like a tuning parameter that could be any `double`. Then even constructing the graph of all the possible targets would be infeasible, let alone building all of them.

I'm sure this situation could be gotten around by generating the set of targets according to user input (so `targets` would never even be given the possibility to construct the branches that the user won't want), but is there a slicker way to do this?

**Comment by wlandau** [comment, marked as answer]
In dynamic branching, the choice of branches is controlled by the upstream target that the downstream target branches over. So if you have `list(tar_target(regions, get_regions()), tar_target(analysis, analyze_region(regions), pattern = map(regions))`, then the user-defined `get_regions()` function controls the choice of regions. If you want to filter down the selected regions, you could build that into the `get_regions()` function or specify it as an argument, or you could have an intermediate target that filters down the specific regions you want: something like `list(tar_target(all_regions, get_all_regions()), tar_target(filtered_regions, filter_regions(all_regions)), tar_target(analysis, analyze_region(filtered_regions), pattern = map(filtred_regions))`.
